src/RadhydroSimulation.hpp

```diff
@@ -259,6 +251,40 @@ template <typename problem_t> class RadhydroSimulation : public AMRSimulation<pr
 			std::array<amrex::MultiFab, AMREX_SPACEDIM> &FOfluxes,	amrex::iMultiFab &redoFlag);
 };
 
+template <typename problem_t>
+void RadhydroSimulation<problem_t>::defineComponentNames() {
+  // check modules cannot be enabled if they are not been implemented yet
+  static_assert(!Physics_Traits<problem_t>::is_chemistry_enabled, "Chemistry is not supported, yet.");
+
+  // cell-centred
+  // add hydro state variables
+  if constexpr (Physics_Traits<problem_t>::is_hydro_enabled ||
+                Physics_Traits<problem_t>::is_radiation_enabled) {
+    std::vector<std::string> hydroNames = {"gasDensity", "x-GasMomentum", "y-GasMomentum", "z-GasMomentum", "gasEnergy", "gasInternalEnergy"};
+    componentNames_cc_.insert(componentNames_cc_.end(), hydroNames.begin(), hydroNames.end());
+    ncomp_cc_ += hydroNames.size();
+  }
+  // add passive scalar variables
+  if constexpr (Physics_Traits<problem_t>::numPassiveScalars > 0){
+    std::vector<std::string> scalarNames = getScalarVariableNames();
+    componentNames_cc_.insert(componentNames_cc_.end(), scalarNames.begin(), scalarNames.end());
+    ncomp_cc_ += scalarNames.size();
+  }
+  // add radiation state variables
+  if constexpr (Physics_Traits<problem_t>::is_radiation_enabled) {
+    std::vector<std::string> radNames = {"radEnergy", "x-RadFlux", "y-RadFlux", "z-RadFlux"};
+    componentNames_cc_.insert(componentNames_cc_.end(), radNames.begin(), radNames.end());
+    ncomp_cc_ += radNames.size();
+  }
+
+  // face-centred
+  // add mhd state variables
+  if constexpr (Physics_Traits<problem_t>::is_mhd_enabled) {
+    componentNames_fc_.push_back({"MagEnergy"});
```

Comment:
You are right, this should not be "MagEnergy" (magnetic energy), because it doesn't have units of energy. Instead,  this is just the amplitude component of the magnetic field vector, in each direction. I have renamed this quantity to "magField", to reflect this.

Now, the reason why there is only a single quantity named, rather than an x-, y-, and possible z-component, is because this quantity will need to be created for each dimension. Here I am just specifying which quantity needs to be defined for magnetic fields to exist. The number of vector components are being interpreted and handled by the code. We can discuss this more tomorrow if you like.